Automattic/wp-calypso
# Welcome Tour: Move the new tour (with step repositioning) behind a feature flag

#### Changes proposed in this Pull Request

Addresses #57056

As of #56867 the Welcome Tour applies step repositioning to move the steps near reference elements on the page (or fallback to the default position - bottom left). It also adds a semi-transparent overlay to block interaction with the editor.

This PR:

- Adds [@automattic/calypso-config](https://www.npmjs.com/package/@automattic/calypso-config) to ETK for using the feature flags API. 
- Moves the core functionality of the "moving tour" in #56867 behind a feature flag `welcome-tour-next`.

#### Testing instructions



1. Sandbox a site, checkout branch, and `yarn dev --sync` from `apps/editing-toolkit` to sync sandbox.
2. Go to the editor, open the sidebar, and from there click on “Welcome Guide”.
3. **In development (sandbox)**: the NEXT welcome tour is enabled by default (dark overlay blocking editor interactions, tour steps moving near reference elements).
4. **In staging (proxied) + flag**: Using the feature flag `?flags=welcome-tour-next` should show the NEXT welcome tour.
5. **In staging (proxied) + no flag**: Not using the feature flag should show the **current** tour: (no dark overlay blocking editor interactions, no tour steps moving near reference elements, the editor behind is interactable and can be clicked to move focus on it).



Closes #57056
